### PR TITLE
Improve docs about read*() functions

### PR DESCRIPTION
**Pull request by nalimilan**
Fix array signatures for readbytes!() and read!().
Add docstring for read(filename::AbstractString, args...).
Mention that text should be UTF-8.
Separate "stream or filename" into two lines with types.
Various minor improvements.

[av skip]

### CODE REVIEW

File: base/docs/helpdb/Base.jl [hunk 1]
```diff
@@ -6307,10 +6325,11 @@ Compute the inverse secant of `x`, where the output is in degrees.
 asecd
 
 """
-    readbytes!(stream, b::Vector{UInt8}, nb=length(b); all=true)
+    readbytes!(stream::IO, b::AbstractVector{UInt8}, nb=length(b); all=true)
 
-Read at most `nb` bytes from the stream into `b`, returning the number of bytes read
-(increasing the size of `b` as needed).
+Read at most `nb` bytes from `stream` into `b`, returning the number of bytes read.
+The size of `b` will be increased if needed (i.e. if `nb` is superior to `length(b)`
```
Comment (tkelman) [inline base/docs/helpdb/Base.jl]: greater than?

**Comment by tkelman** [comment]
Nice cleanup. I've noticed a few other places in the docs that use notation like `foo(a | b)` which could be a bit confusing given that isn't julia type signature syntax

File: base/docs/helpdb/Base.jl (hunk 1 above)
Comment (nalimilan) [inline base/docs/helpdb/Base.jl]: I guess. French influence...

**Comment by nalimilan** [comment]
Indeed. I've added a commit removing these, tell me what you think.

**Comment by tkelman** [comment]
lgtm

**Comment by nalimilan** [comment]
Thanks!

**Pull request merged**